### ISSUE TITLE
[rush] Rush plugin architecture doesn't support ESM?

### ISSUE THREAD

**Issue by kaiyoma** [microsoft/rushstack#4869]
## Summary

I'm working on a custom cloud storage cache plugin, as per the docs here:

https://rushjs.io/pages/maintainer/build_cache/#enabling-cloud-storage
https://rushjs.io/pages/maintainer/using_rush_plugins/

## Repro steps

  **Expected result:** 

  **Actual result:** 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? |  |
| `rushVersion` from rush.json? |  |
| `useWorkspaces` from rush.json? |  |
| Operating system? |  |
| Would you consider contributing a PR? |  |
| Node.js version (`node -v`)? |  |

**Comment by iclanton** [comment]
Yeah, Rush (and actually all of the rushstack projects) currently use the CommonJS module format. Moving to ESM would be a pretty major breaking change, although it may be time to do that. This is something that we'd do with the release of Rush v6, along with other breaking changes and removal of deprecated APIs.

**Comment by NovaAstra** [comment]
> Yeah, Rush (and actually all of the rushstack projects) currently use the CommonJS module format. Moving to ESM would be a pretty major breaking change, although it may be time to do that. This is something that we'd do with the release of Rush v6, along with other breaking changes and removal of deprecated APIs.

When is Rush v6 expected to be released?
